psychopy/sound/_base.py

```diff
@@ -137,7 +137,7 @@ def _parseSpeaker(self, speaker):
 
         return device
 
-    def setSound(self, value, secs=0.5, octave=4, hamming=True, log=True):
+    def setSound(self, value, device, secs=0.5, octave=4, hamming=True, log=True):
```

Comment:
I'm concerned adding `device` here may break existing code that doesn't set that value. Possible solution would be to append `device` to the end of the keyword arguments and allow it to take None.